packages/kit/src/utils/routing.js

```diff
@@ -21,7 +21,8 @@ export function parse_route_id(id) {
 			: new RegExp(
 					`^${get_route_segments(id)
 						.map((segment, i, segments) => {
-							const decoded_segment = decodeURIComponent(segment);
+							// exclude %25 from decoding to match decoding of url.pathname
+							const decoded_segment = segment.split('%25').map(decodeURIComponent).join('%25');
```

Comment:
this seems like it's probably broken. `decode_pathname` uses `decodeURI`, which I think it correct. we shouldn't be doing it a different way here

Reply:
Yes, it seems that avoiding %25 doesn't solve all problems.